src/Cms/FilePicker.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+namespace Kirby\Cms;
+
+use Kirby\Exception\InvalidArgumentException;
+
+/**
+ * The FilePicker class helps to
+ * fetch the right files for the API calls
+ * for the file picker component in the panel.
+ *
+ * @package   Kirby Cms
+ * @author    Bastian Allgeier <bastian@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier GmbH
+ * @license   https://getkirby.com/license
+ */
+class FilePicker
+{
+    /**
+     * @var \Kirby\Cms\App
+     */
+    protected $kirby;
+
+    /**
+     * @var array
+     */
+    protected $options;
+
+    /**
+     * @var \Kirby\Cms\Site
+     */
+    protected $site;
+
+    /**
+     * Creates a new FilePicker instance
+     *
+     * @param array $params
+     */
+    public function __construct(array $params = [])
+    {
+        // default params
+        $defaults = [
+            // image settings (ratio, cover, etc.)
+            'image' => [],
+            // query template for the file info field
+            'info' => false,
+            // number of pages displayed per pagination page
+            'limit' => 20,
+            // optional mapping function for the pages array
+            'map' => null,
+            // the reference model (site, page, file or user)
+            'model' => site(),
+            // current page when paginating
+            'page' => 1,
+            // a query string to fetch specific pages
+            'query' => null,
+            // query template for the file text field
+            'text' => '{{ file.filename }}'
+        ];
+
+        $this->options = array_merge($defaults, $params);
+        $this->kirby   = $this->options['model']->kirby();
+        $this->site    = $this->kirby->site();
+    }
+
+
+    /**
+     * Search all files for the picker
+     *
+     * @return \Kirby\Cms\Files|null
+     */
+    public function files()
+    {
+        $model = $this->options['model'];
+
+        // find the right default query
+        if (empty($this->options['query']) === false) {
+            $query = $this->options['query'];
+        } elseif (is_a($model, 'Kirby\Cms\File') === true) {
+            $query = 'file.siblings';
+        } else {
+            $query = $model::CLASS_ALIAS . '.files';
+        }
+
+        // fetch all files for the picker
+        $files = $model->query($query);
+
+        // help mitigate some typical query usage issues
```

Comment:
Does this make sense in the context of the FilePicker? Shouldn't it covert to files not children?